src/components/textbox/textbox.test.tsx

```diff
@@ -0,0 +1,607 @@
+      const validationIcon = screen.getByTestId(`icon-${type}`);
+      expect(inputPresentationContainer).toContainElement(validationIcon);
+    }
+  );
+
+  test.each(validationTypes)(
+    `when %s prop passed as string and validationOnLabel
+    as true render proper validation icon on the label`,
+    (type) => {
+      render(
+        <Textbox label="Label" {...{ [type]: "Message" }} validationOnLabel />
+      );
+      const labelContainer = screen.getByTestId("label-container");
+      const validationIcon = screen.getByTestId(`icon-${type}`);
+      expect(labelContainer).toContainElement(validationIcon);
+    }
+  );
+
+  test.each([
+    ["top", true],
+    ["bottom", true],
+    ["left", true],
+    ["top", false],
+    ["bottom", false],
+    ["left", false],
+  ] as const)(
+    "should pass the expected value %s rather than the default ('right') when overriding the tooltip position",
+    (tooltipPosition, onLabel) => {
+      const useFloatingSpy = jest.spyOn(floatingUi, "useFloating");
+
+      render(
+        <Textbox
+          label="Label"
+          error="Message"
+          validationOnLabel={onLabel}
+          tooltipPosition={tooltipPosition}
+        />
+      );
+
+      expect(useFloatingSpy).toHaveBeenCalledWith(
+        expect.objectContaining({ placement: tooltipPosition })
+      );
+
+      useFloatingSpy.mockRestore();
+    }
+  );
+});
+
+describe("when the prefix prop is set", () => {
+  test("then a StyledPrefix should be rendered with this prop value", () => {
+    const prefixValue = "bar";
+    render(<Textbox value="foo" prefix={prefixValue} />);
+    expect(screen.getByText(prefixValue)).toHaveAttribute(
+      "data-element",
+      "textbox-prefix"
+    );
+  });
+
+  test("when the align prop is 'right' then 'flex-direction' should be 'row'", () => {
+    const prefixValue = "bar";
+    render(<Textbox value="foo" prefix={prefixValue} align="right" />);
+    expect(screen.getByRole("presentation")).toHaveStyle({
+      flexDirection: "row",
+    });
+  });
+});
+
+describe("required", () => {
+  test("the required prop is passed to the input", () => {
+    render(<Textbox value="foo" label="Required" required />);
+    expect(screen.getByRole("textbox")).toBeRequired();
+  });
+
+  test("the label includes the 'required' asterisk", () => {
+    render(<Textbox value="foo" label="Required" required />);
+    expect(screen.getByText("Required")).toHaveStyleRule("content", '"*"', {
+      modifier: "::after",
+    });
+  });
+});
+
+test("renders the positionChildren prop before the input", () => {
+  const Component = () => <div>positionedChildren content</div>;
+  render(<Textbox positionedChildren={<Component />} />);
+  const positionChildren = screen.getByText("positionedChildren content");
+  const input = screen.getByRole("textbox");
+  expect(positionChildren.compareDocumentPosition(input)).toEqual(
+    Node.DOCUMENT_POSITION_FOLLOWING
+  );
+});
+
+describe("aria attributes", () => {
```

Comment:
suggestion(non-blocking): this describe block isn't adding much value

We should probably either be more verbose with the description or drop the describe block.